### PR TITLE
Feature/crd 48

### PR DESCRIPTION
**Pull request by CrazyPandabier**

### CODE REVIEW

File: cordyceps/cordyceps/vs_controller.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+        # Constants delcaration
+        self.MAX_BOT_SPEED =  0.5 #m/s
+        self.GOAL_RADIUS = 0.05 #m
+
+        self.robots = []
+        for i in range(fleet_size):
+            self.robots.append(Robot(0,0,0,f"r{i}", self))
+
+        self.follow_paths_thread = None
```
Comment (gaspatxo) [inline cordyceps/cordyceps/vs_controller.py:27]:
```suggestion
        self.follow_paths_thread = None # wtf
```

File: cordyceps/cordyceps/vs_controller.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
+import rclpy
+from rclpy.node import Node
+import numpy as np
+import matplotlib.pyplot as plt
+from geometry_msgs.msg import Twist
+import threading
+from .Robot import Robot
+from std_srvs.srv import Trigger
+from cordyceps_interfaces.msg import RobotPaths
+
```
Comment (gaspatxo) [inline cordyceps/cordyceps/vs_controller.py]:
```suggestion
```

File: cordyceps/cordyceps/vs_controller.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+from cordyceps_interfaces.srv import Controller, CheckThread
```
Comment (gaspatxo) [inline cordyceps/cordyceps/vs_controller.py]:
```suggestion
from cordyceps_interfaces.msg import RobotPaths
from cordyceps_interfaces.srv import Controller, CheckThread
```

**Review by SaraEersel — APPROVED** [review]

File: cordyceps/cordyceps/vs_controller.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        self.follow_paths_thread.start()
+        return response
+    
+    def check_thread_state_callback(self, request, response):
+        """Checks if the thread is still running"""
+        response.is_alive = self.follow_paths_thread.is_alive()
+        return response
+
+    def plot_path(self, robot_paths:list[list[tuple[float, float]]]) -> None:
+        labels = []
```
Comment (gaspatxo) [inline cordyceps/cordyceps/vs_controller.py:51]:
```suggestion
    def plot_path(self, robot_paths:list[list[tuple[float, float]]]) -> None:
    """Displays the current path"""
    
```

Comment (SaraEersel) [reply cordyceps/cordyceps/vs_controller.py:51]: labels[] needs to stay there because it is used in legend

**Review by SaraEersel — CHANGES_REQUESTED** [review]

**Review by SaraEersel — APPROVED** [review]

**Pull request merged**